modules/homa-lite/src/lib.rs

```diff
@@ -257,14 +261,31 @@ pub mod module {
 						debug_assert!(res.is_ok());
 
 						if res.is_ok() {
-							current_weight = required_weight;
+							current_weight = <T as Config>::WeightInfo::xcm_unbond();
 
 							scheduled_unbond.remove(0);
 							ScheduledUnbond::<T>::put(scheduled_unbond);
 						}
 					}
 				}
 			}
+
+			// With remaining weight, calculate max number of redeems that can be matched
+			let num_redeem_matches = remaining_weight
+				.saturating_sub(current_weight)
+				.checked_div(<T as Config>::WeightInfo::redeem_with_available_staking_balance())
+				.unwrap_or_default();
+
+			// Iterate through existing redeem_requests, and try to match them with `available_staking_balance`
+			let res = Self::process_redeem_requests_with_available_staking_balance(num_redeem_matches);
+			debug_assert!(res.is_ok());
+			if res.is_ok() {
```

Comment:
```suggestion
			if Ok(count) = res) {
```
and replace `res.unwrap_or_default()`